Source/Particles/ParticleBoundaries.H

```diff
@@ -0,0 +1,141 @@
+/* Copyright 2021 David Grote
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef PARTICLEBOUNDARIES_H_
+#define PARTICLEBOUNDARIES_H_
+
+#include "Utils/WarpXAlgorithmSelection.H"
+
+#include <AMReX_AmrCore.H>
+
+/**
+ *  This allows a different boundary condition on each edge of the grid.
+ */
+struct ParticleBoundaries
+{
+
+    ParticleBoundaries () noexcept
```

Comment:
Just saw this: in order to keep compile time for TUs that include this header low, we should probably move all those implementations into a `.cpp` file.

Reply:
I can submit a PR to do this.